apps/meteor/client/views/room/UserCard/UserCardWithData.tsx

```diff
@@ -24,7 +25,16 @@ const UserCardWithData = ({ username, rid, onOpenUserInfo, onClose }: UserCardWi
 	const getRoles = useRolesDescription();
 	const showRealNames = Boolean(useSetting('UI_Use_Real_Name'));
 
-	const { data, isLoading } = useUserInfoQuery({ username });
+	const { data, isLoading: isUserInfoLoading } = useUserInfoQuery({ username });
+	const {
+		data: isMemberData,
+		refetch,
+		isSuccess: membershipCheckSuccess,
+		isLoading: isMembershipStatusLoading,
+	} = useMemberExists({ roomId: rid, username });
+
+	const isLoading = isUserInfoLoading || isMembershipStatusLoading;
+	const isMember = (isMemberData?.exists as boolean) && membershipCheckSuccess;
```

Comment:
If you first check for success I believe you don't need to cast this. @gabriellsh can you help us here?